### PR TITLE
ui: gallery enhancements

### PR DESCRIPTION
**Pull request by psychedelicious**
huge slew of gallery related improvements accumulated while attempting to figure out the image fetching stuff (complications fetching boards)

@maryhipp 

- Remove lightbox entirely (it is so far behind, needs to be totally redone anyways sry @blessedcoolant )
- Fix dnd sensor to allow immediate drag and drop
- Fix dnd to accept array of image names as payload
- Add action denylist for redux devtools that filters out RTK Query internal actions
- Consolidate batch and gallery slices - `batchImageNames` is now on gallery
- Performance/rerender enhancements, especially related to dnd and gallery
- Fix up skeletons to work more better
- Make IAIDndImage fallback use the thumbnail URL when loading the image, much faster
- Consolidate/modularize Board & Gallery, ImageContextMenu, ImageMetadata components

### CODE REVIEW

**Comment by psychedelicious** [comment]
Some things aren't working:
- The network requests associated with board actions are mostly not working. The function calls are still in there but the backend and actual queries don't exist.
- Edge cases with multi-select and deleting images/removing images from batches

General ~nodes~ excuse me, *notes*:
- The `BoardsList` has buttons for `All Images`, `Batch` and each `User Board`. `No Board` should also be there. Or, at least, these are the ~board-like entities we should expose
- Selection is an array of image names in the gallery. This allows any collection of images to be displayed in the gallery pane and still get multi-select on them. It may be that this logic should be separate from the gallery, like just in a hook. It shouldn't be persisted.
- Drag a multiple selection and it has a payload of the image names of the selection.
- Batch is an array of image names. I've removed all references to batch except for its button in the `BoardsList`. It doesn't do anything now.

The backend changes to support the network requests are here: https://github.com/invoke-ai/InvokeAI/compare/main...feat/ui/gallery-enhancements

I split them out bc I'm not sure if they are the right approach, but check the commit message for details on the changes there.

**Review by maryhipp — APPROVED** [review]
yolo

**Comment by maryhipp** [comment]
the state of this is "usable enough to go into main" but lets chat about whether now is the time to unify the API calls for images and board_images (as a quick follow)

File: invokeai/frontend/web/src/features/gallery/components/ImageGrid/GalleryImageGrid.tsx [hunk 1]
```diff
@@ -82,20 +108,6 @@ const GalleryImageGrid = () => {
...
-      initialize({
-        target: root,
-        elements: {
-          viewport: scroller,
-        },
-      });
-    }
-    return () => osInstance()?.destroy();
-  }, [scroller, initialize, osInstance]);
-
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/gallery/components/ImageGrid/GalleryImageGrid.tsx]: @maryhipp this sets up the custom scrollbars, any reason to remove it?

**Comment by maryhipp** [comment]
Nope I think I thought it was related to the endless scrolling --
Mary Hipp Rogers
***@***.***
(770)856-7325

**Pull request merged**